kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/BackoffStrategy.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption;
+
+import java.time.Duration;
+
+public interface BackoffStrategy {
```

Comment:
I wonder if this might end up being something with utility outside encryption, but I'm happy that we wait and see what emerges as we get into other use-cases.

Reply:
Yeah I think so, but needs some thought, similar to the thought about whether having some `ByteBuffer` manipulation classes somewhere common would be useful to Filter authors. I'm cautious about having some `commons-lang` type lib that could end up being used in Filters and the Proxy implementation with all these things depending on it's APIs. 

Maybe we could have a lib named like `filter-support`, it could be targeted at Filter Authors and we would specify that it's not on the classpath of Kroxy by default but is to be user managed.